packages/@expo/cli/src/start/server/BundlerDevServer.ts

```diff
@@ -403,6 +410,8 @@ export abstract class BundlerDevServer {
   protected shouldUseInterstitialPage(): boolean {
     return (
       env.EXPO_ENABLE_INTERSTITIAL_PAGE &&
+      // if user passed --dev-client flag, skip interstitial page
+      !this.isDevClient &&
```

Comment:
If the user passed in `--dev-client` then this function should never be called, `openAsync` should use the `custom` option.

Reply:
Without this here, the QR code would now point to interstitial page when `--dev-client` is passed.